### PR TITLE
Chore/python clean up, and unused variable check to pre commit

### PR DESCRIPTION
**Pull request by j1m-ryan**
### Proposed changes

- removed an obscured function
- adding pre commit tooling to error on unused imports/ unused variables.
- autopep8 format

### CODE REVIEW

**Review by jjngx — APPROVED** [review]

**Review by haywoodsh — APPROVED** [review]

File: tests/suite/fixtures/ic_fixtures.py [hunk 1]
```diff
@@ -267,7 +267,7 @@ def crd_ingress_controller_with_dos(
 
         print("------------------------- Create syslog svc -----------------------")
         src_syslog_yaml = f"{TEST_DATA}/dos/dos-syslog.yaml"
-        log_loc = f"/var/log/messages"
+        f"/var/log/messages"
```
Comment (pdabelf5) [inline tests/suite/fixtures/ic_fixtures.py:270]: is this needed at all?

File: tests/suite/test_annotations.py [hunk 2]
```diff
@@ -566,7 +566,7 @@ def test_standard_ingress(
         unexpected_strings,
     ):
         initial_events = get_events(kube_apis.v1, annotations_setup.namespace)
-        initial_count = get_event_count(annotations_setup.ingress_event_text, initial_events)
+        get_event_count(annotations_setup.ingress_event_text, initial_events)
```
Comment (pdabelf5) [inline tests/suite/test_annotations.py:569]: This whole line can be removed?

Comment (j1m-ryan) [reply tests/suite/test_annotations.py]: good catch.  And the one above as initial events is not used elsewhere. I am assuming these get functions have no side effects, but we will know when the tests run

File: tests/suite/fixtures/ic_fixtures.py (hunk 1 above)
Comment (j1m-ryan) [reply tests/suite/fixtures/ic_fixtures.py]: Good catch. no its not, I have removed it now. Thanks

**Pull request merged**